### ISSUE TITLE
Unclear suffix naming convention with duplicate operation ID's

### ISSUE THREAD

**Issue by marcrd** [wcandillon/swagger-js-codegen#192]
In https://github.com/wcandillon/swagger-js-codegen/pull/186, it was changed so if you have duplicate operation ID's, the library appends numeric suffixes using `_1`, `_2`, etc. 

These numerically suffixed methods don't provide much context to the caller. I'd propose appending the path parameters to the method name, so as to make it unique with the parameter context.

For example, let's say you had multiple operation ID's of `ListAggregatedUsers`, with different path parameters. Instead of the numerically generated suffixes, append that URL parameters of say `resourceID` to the operation ID for the client. So instead of:

```
ListAggregatedUsers_1
```

Becomes:

```
ListAggregatedUsersWithResourceId
```

**Comment by marcrd** [comment]
Bumping this

**Comment by HawkeyePierce89** [comment]
Yes, it's bad solution :( And this works wihtout reference to ID's, just in the order of methods in Swagger

**Comment by HawkeyePierce89** [comment]
How a temporary solution you can use ```getPathToMethodName(opts, m, path))``` instead ```var methodName = (op.operationId ? normalizeName(op.operationId) : getPathToMethodName(opts, m, path));```

**Comment by marcrd** [comment]
> How a temporary solution you can use getPathToMethodName(opts, m, path)) instead var methodName = (op.operationId ? normalizeName(op.operationId) : getPathToMethodName(opts, m, path));

That's a good idea, I'll give it a try!
